crates/wit-encoder/src/result.rs

```diff
@@ -0,0 +1,60 @@
+use std::fmt::Display;
+
+use crate::Type;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Result_ {
+    ok: Option<Type>,
+    err: Option<Type>,
+}
+
+impl Result_ {
+    pub fn ok(type_: Type) -> Self {
+        Self {
+            ok: Some(type_),
+            err: None,
+        }
+    }
+    pub fn err(type_: Type) -> Self {
+        Self {
+            ok: None,
+            err: Some(type_),
+        }
+    }
+    pub fn both(ok: Type, err: Type) -> Self {
+        Self {
+            ok: Some(ok),
+            err: Some(err),
+        }
+    }
+    pub fn empty() -> Self {
+        Self {
+            ok: None,
+            err: None,
+        }
+    }
+    pub fn is_empty(&self) -> bool {
+        self.ok.is_none() && self.err.is_none()
+    }
+}
+
+impl Display for Result_ {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "result")?;
+        if !self.is_empty() {
+            write!(f, "<")?;
+            if let Some(type_) = &self.ok {
+                type_.fmt(f)?;
+            }
+            if let Some(type_) = &self.err {
+                if self.ok.is_none() {
+                    write!(f, "_")?;
+                }
```

Comment:
Maybe this write! could be moved to an else clause for the earlier 'if let' on self.ok? It might read a little better but maybe I'm missing something. 

I could guess this was written before the '!self.is_empty()` was added.

Reply:
Good catch @FrankReh!
I've updated it.